File: backend/models/Favorito.js
const pool = require('../config/database');

class Favorito {
  static async findByUsuarioId(usuario_id) {
    const result = await pool.query(
      `SELECT f.*, 
              COALESCE(a.titulo, m.titulo) as titulo,
              COALESCE(a.imagen_url, m.imagen_portada) as imagen
       FROM favoritos f
       LEFT JOIN animes a ON f.anime_id = a.id
       LEFT JOIN mangas m ON f.manga_id = m.id
       WHERE f.usuario_id = $1`,
      [usuario_id]
    );
    return result.rows;
  }

  static async findById(id) {
    const result = await pool.query('SELECT * FROM favoritos WHERE id = $1', [id]);
    return result.rows[0];
  }

  static async findByUsuarioAndAnime(usuario_id, anime_id) {
    const result = await pool.query(
      'SELECT * FROM favoritos WHERE usuario_id = $1 AND anime_id = $2',
      [usuario_id, anime_id]
    );
    return result.rows[0];
  }

  static async findByUsuarioAndManga(usuario_id, manga_id) {
    const result = await pool.query(
      'SELECT * FROM favoritos WHERE usuario_id = $1 AND manga_id = $2',
      [usuario_id, manga_id]
    );
    return result.rows[0];
  }

  static async create({ usuario_id, anime_id = null, manga_id = null }) {
    const result = await pool.query(
      'INSERT INTO favoritos (usuario_id, anime_id, manga_id) VALUES ($1, $2, $3) RETURNING *',
      [usuario_id, anime_id, manga_id]
    );
    return result.rows[0];
  }

  static async delete(id) {
    await pool.query('DELETE FROM favoritos WHERE id = $1', [id]);
  }

  static async deleteByUsuarioAndAnime(usuario_id, anime_id) {
    await pool.query(
      'DELETE FROM favoritos WHERE usuario_id = $1 AND anime_id = $2',
      [usuario_id, anime_id]
    );
  }

  static async deleteByUsuarioAndManga(usuario_id, manga_id) {
    await pool.query(
      'DELETE FROM favoritos WHERE usuario_id = $1 AND manga_id = $2',
      [usuario_id, manga_id]
    );
  }
}

module.exports = Favorito;

File: models/favorito.py
from models.database import pool

class Favorito:
    @staticmethod
    def find_by_usuario_id(usuario_id):
        result = pool.execute(
            """SELECT f.id, f.usuario_id, f.anime_id, f.manga_id, f.fecha_agregado,
                      COALESCE(a.titulo, m.titulo) as titulo,
                      COALESCE(a.imagen_url, m.imagen_portada) as imagen
             FROM favoritos f
             LEFT JOIN animes a ON f.anime_id = a.id
             LEFT JOIN mangas m ON f.manga_id = m.id
             WHERE f.usuario_id = %s""",
            [usuario_id]
        )
        return [{'id': r[0], 'usuario_id': r[1], 'anime_id': r[2], 'manga_id': r[3], 'titulo': r[5], 'imagen': r[6]} for r in result]
    
    @staticmethod
    def find_by_usuario_and_anime(usuario_id, anime_id):
        result = pool.execute(
            'SELECT * FROM favoritos WHERE usuario_id = %s AND anime_id = %s',
            [usuario_id, anime_id]
        )
        return result[0] if result else None
    
    @staticmethod
    def find_by_usuario_and_manga(usuario_id, manga_id):
        result = pool.execute(
            'SELECT * FROM favoritos WHERE usuario_id = %s AND manga_id = %s',
            [usuario_id, manga_id]
        )
        return result[0] if result else None
    
    @staticmethod
    def create(usuario_id, anime_id=None, manga_id=None):
        result = pool.execute(
            'INSERT INTO favoritos (usuario_id, anime_id, manga_id) VALUES (%s, %s, %s) RETURNING *',
            [usuario_id, anime_id, manga_id]
        )
        return {'id': result[0][0], 'usuario_id': result[0][1], 'anime_id': result[0][2], 'manga_id': result[0][3]}
    
    @staticmethod
    def delete(favorito_id):
        pool.execute('DELETE FROM favoritos WHERE id = %s', [favorito_id])
    
    @staticmethod
    def delete_by_usuario_and_anime(usuario_id, anime_id):
        pool.execute('DELETE FROM favoritos WHERE usuario_id = %s AND anime_id = %s', [usuario_id, anime_id])
    
    @staticmethod
    def delete_by_usuario_and_manga(usuario_id, manga_id):
        pool.execute('DELETE FROM favoritos WHERE usuario_id = %s AND manga_id = %s', [usuario_id, manga_id])